File: app/src/main/java/com/android/sellacha/adapter/NotificationAdapter.java
package com.android.sellacha.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.text.format.DateUtils;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.android.sellacha.R;
import com.android.sellacha.databinding.NotificationItemBinding;

import java.util.List;

public class NotificationAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    Context context;
    List<NotificationResponse> notificationList;

    public NotificationAdapter(Context context, List<NotificationResponse> notificationList) {
        this.context = context;
        this.notificationList = notificationList;
    }

    private RecyclerView.ViewHolder getViewHolder(LayoutInflater inflater, ViewGroup group) {
        NotificationItemBinding binding = DataBindingUtil.inflate(inflater, R.layout.notification_item, group, false);
        return new NotificationHolder(binding);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return getViewHolder(LayoutInflater.from(context), parent);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, @SuppressLint("RecyclerView") int position) {
        NotificationAdapter.NotificationHolder vHolder = (NotificationAdapter.NotificationHolder) holder;
        vHolder.binding.notificationTxt.setText(notificationList.get(position).getMessage());
        vHolder.binding.notificationTime.setText(notificationList.get(position).getDate());

    }

    @Override
    public int getItemCount() {
        return notificationList.size();
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    public static class NotificationHolder extends RecyclerView.ViewHolder {
        NotificationItemBinding binding;

        public NotificationHolder(NotificationItemBinding itemView) {
            super(itemView.getRoot());
            binding = itemView;
        }

    }
}

File: app/src/main/java/com/android/sellacha/Notification/Adapter/NotificationAdapter.kt
package com.android.sellacha.Notification.Adapter

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.android.sellacha.R
import com.android.sellacha.databinding.NotificationItemBinding

class NotificationAdapter(var context: Context, var notificationList: List<Any>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    private fun getViewHolder(inflater: LayoutInflater, group: ViewGroup): RecyclerView.ViewHolder {
        val binding = DataBindingUtil.inflate<NotificationItemBinding>(
            inflater,
            R.layout.notification_item,
            group,
            false
        )
        return NotificationHolder(binding)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return getViewHolder(LayoutInflater.from(context), parent)
    }

    override fun onBindViewHolder(
        holder: RecyclerView.ViewHolder,
        @SuppressLint("RecyclerView") position: Int
    ) {
        val vHolder = holder as NotificationHolder
//        vHolder.binding.notificationTxt.text = notificationList[position].data.toString()
//        vHolder.binding.notificationTime.text = notificationList[position].date
    }

    override fun getItemCount(): Int {
        return notificationList.size
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }

    class NotificationHolder(var binding: NotificationItemBinding) : RecyclerView.ViewHolder(
        binding.root
    )
}